examples/main/main.cpp

```diff
@@ -865,9 +865,20 @@ int main(int argc, char ** argv) {
                 console::set_display(console::reset);
                 display = true;
 
-                // Add tokens to embd only if the input buffer is non-empty
-                // Entering a empty line lets the user pass control back
-                if (buffer.length() > 1) {
+                if (!buffer.length()) { // Ctrl+D on empty line exits
```

Comment:
```suggestion
                if (buffer.empty()) { // Ctrl+D on empty line exits
```

Does this also work if user hit enter (empty line)?

Reply:
The existing behavior of instead letting "the user pass control back" works. I preserved the behavior from #571 on line 881 below. Enter and Ctrl+D should, and now do, do different things. At this line, they are represented in memory as `buffer == "\n"` and `buffer == ""` respectively.